src/ralph/assets/api/serializers.py

```diff
@@ -437,9 +437,18 @@ class ComponentSerializerMixin(NetworkComponentSerializerMixin):
     processors = ProcessorSimpleSerializer(many=True, source='processor_set')
 
 
+class SecurityScanField(serializers.Field):
+
+    def to_representation(self, value):
+        if getattr(value, 'pk'):
+            return SecurityScanSerializer().to_representation(value)
+        else:
```

Comment:
what is (type of) value here? could we reverse this condition into sth like:
```
if not value:  # or if value is None
    return None
else:
    return SecurityScanSerializer().to_representation(value)
```

Reply:
It could be None or a SecurityScan instance. I will add an extra check at the top to see if it is not none and then do as you say.